service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/DateFormatUtil.java

```diff
@@ -1,26 +1,73 @@
 package uk.nhs.adaptors.gp2gp.ehr.utils;
 
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.DAY;
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.MINUTE;
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.MONTH;
+import static ca.uhn.fhir.model.api.TemporalPrecisionEnum.YEAR;
+
+import ca.uhn.fhir.model.api.TemporalPrecisionEnum;
+
+import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
-import java.time.format.DateTimeFormatterBuilder;
-import java.util.Date;
+import java.util.Optional;
+
+import org.hl7.fhir.dstu3.model.DateTimeType;
+import org.hl7.fhir.dstu3.model.DateType;
+import org.hl7.fhir.dstu3.model.InstantType;
+
+import com.google.common.collect.ImmutableMap;
 
 import uk.nhs.adaptors.gp2gp.ehr.exception.EhrMapperException;
 
 public class DateFormatUtil {
-    private static final String UK_ZONE_ID = "Europe/London";
-    private static final DateTimeFormatter DATE_TIME_FORMATTER = new DateTimeFormatterBuilder()
-        .appendPattern("yyyyMMddHHmmss")
-        .toFormatter();
-
-    public static String formatDate(Date date) {
-        if (date == null) {
-            throw new EhrMapperException("Could not format date");
+
+    public static final String UK_TIMEZONE = "Europe/London";
+    public static final String COULD_NOT_FORMAT_DATE = "Could not format date";
+    private static final ImmutableMap<TemporalPrecisionEnum, String> YEAR_PATTERN_MAP = ImmutableMap.of(
+        YEAR, "yyyy",
```

Comment:
since DateTimeFormatter is threadsafe and immutable you could consider creating the formatter here and reusing it instead of creating a new formatter from its string representation every time.